misk/src/test/kotlin/misk/web/actions/ResponseContentTypeTest.kt

```diff
@@ -0,0 +1,102 @@
+package misk.web.actions
+
+import com.squareup.protos.test.grpc.HelloReply
+import misk.MiskTestingServiceModule
+import misk.inject.KAbstractModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.web.Get
+import misk.web.PathParam
+import misk.web.ResponseContentType
+import misk.web.WebActionModule
+import misk.web.WebServerTestingModule
+import misk.web.jetty.JettyService
+import misk.web.mediatype.MediaTypes
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.nio.charset.Charset
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@MiskTest(startService = true)
+class ResponseContentTypeTest {
+  @MiskTestModule
+  val module = TestModule()
+
+  @Inject private lateinit var jettyService: JettyService
+
+  private val httpClient = OkHttpClient()
+
+  @Test
+  fun `the Accept header can control what response type is returned if the endpoint supports it`() {
+    val requestForJson = get("/hello/jeff", MediaTypes.APPLICATION_JSON_MEDIA_TYPE)
+    val jsonResponse = httpClient.newCall(requestForJson).execute()
+    assertThat(jsonResponse.code).isEqualTo(200)
+    assertThat(jsonResponse.body!!.string()).isEqualTo("""{"message":"howdy, jeff"}""")
+
+    val requestForProto = get("/hello/jeff", MediaTypes.APPLICATION_PROTOBUF_MEDIA_TYPE)
+    val protoResponse = httpClient.newCall(requestForProto).execute()
+    assertThat(protoResponse.code).isEqualTo(200)
+    assertThat(protoResponse.body!!.string())
+      .isEqualTo(HelloReply("howdy, jeff").encodeByteString().string(Charset.defaultCharset()))
+  }
+
+  @Test
+  fun `the Accept header does not allow unsupported types to be returned`() {
```

Comment:
Should we have a test for no accept header on an action which has multiple return types? ie. the `/hello/` proto/json endpoint and confirm a deterministic default return type is selected?